rewrite-pattern.js

```diff
@@ -452,7 +454,38 @@ const processCharacterClass = (
 			if (config.useUnicodeFlag) {
 				update(characterClassItem, `[^${setStr[0] === '[' ? setStr.slice(1, -1) : setStr}]`)
 			} else {
-				update(characterClassItem, `(?!${setStr})[\\s\\S]`)
+				if (config.flags.unicode) {
+					if (config.flags.ignoreCase) {
+						const astralCharsSet = singleChars.clone().intersection(ASTRAL_SET);
+						// Assumption: singleChars do not contain lone surrogates.
+						// Regex like /[^\ud800]/u is not supported
```

Comment:
Maybe we should throw if we find an unsupported regex?

Reply:
There is not precedence, we don't throw on valid regex and `/[^\ud800]/u` is valid for sure.

If we want to support lone surrogate in a unicode regex, we could resort to the `UNICODE_SET \ singleChars` approach, but then we lost the `ignoreCase` folding expansion.